### DISCUSSION TITLE
GitHub Contributor

### DISCUSSION THREAD

**Discussion by Jayantadeb73** [community/community#21737] [New to GitHub]
<p>Hi All,</p>
<p>It will be nice if any one can explain with real life senario, when we should add an github user as a ‘Contributor’ other than ‘Collaborator’ ?</p>

**Comment by Gisgar3** [comment, marked as answer]
<p>In a real-life situation, it is best suited that you would make a person a  <strong>Contributor</strong> if they are a valuable asset to the code-base and are dedicated to the project at hand. Normal  <strong>Collaborators</strong> are usually people that make small changes and are not highly necessary for overall code peformance.</p>

**Comment by lee-dohm** [comment]
<p>Interesting question!</p>
<p>Let’s start with some definitions:</p>
<ul>
<li>A “contributor” is someone who has, at any time in the past, had code merged into your repository</li>
<li>
<a href="https://help.github.com/en/articles/inviting-collaborators-to-a-personal-repository">A “collaborator”</a> is someone who has been granted more than default access to your repository</li>
</ul>
<p>GitHub will add a badge to the comments of anyone who is a contributor to your repository automatically, see <a href="https://github.com/atom/atom/issues/19174">this issue in atom/atom</a> for example:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e5eecc0897b0ba7905d5f9cfda66d3ab6dbd9afc.png" data-download-href="/uploads/short-url/wO4XzVp0cVcs6QclNZgKYJ3rgFm.png?dl=1" title="Screen Shot 2019-05-15 at 8.27.09 AM.png"><img src="https://user-images.githubusercontent.com/1038121/181066485-57062362-ca02-476f-8eea-7fd11648a9b7.png" alt="Screen Shot 2019-05-15 at 8.27.09 AM.png" data-base62-sha1="wO4XzVp0cVcs6QclNZgKYJ3rgFm" width="690" height="214" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e5eecc0897b0ba7905d5f9cfda66d3ab6dbd9afc_2_690x214.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e5eecc0897b0ba7905d5f9cfda66d3ab6dbd9afc_2_1035x321.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e5eecc0897b0ba7905d5f9cfda66d3ab6dbd9afc_2_1380x428.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e5eecc0897b0ba7905d5f9cfda66d3ab6dbd9afc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-05-15 at 8.27.09 AM.png</span><span class="informations">1778×554 77.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You don’t have to do anything for that to happen, other than merge their code. A collaborator is a more manual process because you’re trusting them on an ongoing basis by granting them write access to your repository.</p>
<p>I hope that helps and let us know if you have more questions.</p>

**Comment by Jayantadeb73** [comment]
<p>Hi Gisgar,</p>
<p>Thanks for your explanation.</p>

**Comment by Jayantadeb73** [comment]
<p>Hi lee-dohm,</p>
<p>Nice to see your great explanation. </p>
